### PR TITLE
feat(pdk) kong.service.response.get_upstream

### PR DESCRIPTION
**Pull request by fairyqb**
### Summary

PDK: 
**kong.service.response.get_upstream**
return upstream server IP address and port or the path to the UNIX-domain socket

The upstream service has multiple target nodes, when one of them has a business problem, it is useful to quickly resolve the issue by returning the response header to the client.

### CODE REVIEW

**Comment by fairyqb** [comment]
My local test was successful, but the server check failed? Why?

**Comment by fairyqb** [comment]
The Travis CI build failed  ??

2020/09/02 00:43:17 [error] 20583#0: *2 [kong] workspaces.lua:12 [off] cannot create 'workspaces' entities when not using a database, context: ngx.timer

File: kong/pdk/service/response.lua [hunk 1]
```diff
@@ -136,6 +139,28 @@ local function new(pdk, major_version)
...
+  -- @treturn string|nil the ip:port from the response from the Service, or `nil`
+  -- @usage
+  -- kong.log.inspect(kong.service.response.get_upstream()) -- 127.0.0.1:80
+  function response.get_upstream()
+    check_phase(header_body_log)
+
+    if ngx.var.upstream_addr then
+      local addrs = split(ngx.var.upstream_addr, ',')
+      if #addrs > 0 then
+          return strip(addrs[#addrs])
```
Comment (kikito) [inline kong/pdk/service/response.lua:156]: This needs a little bit more. According to http://nginx.org/en/docs/http/ngx_http_upstream_module.html ,

> If an internal redirect from one server group to another happens, initiated by `X-Accel-Redirect` or error_page, then the server addresses from different groups are separated by colons, e.g. `192.168.1.1:80, 192.168.1.2:80, unix:/tmp/sock : 192.168.10.1:80, 192.168.10.2:80`

My suggestion is also splitting by `:`:

```suggestion
      addrs = split(addrs, ':')
      if #addrs > 0 then
        return strip(addrs[#addrs])]
      end
```

**Comment by kikito** [comment]
I'm closing this due to lack of activity. Please reopen if needed. Thanks for taking the time to send a PR!

**Pull request closed without merging**